### PR TITLE
doc: release: 3.5: Add note on RaspberryPi Pico related changes

### PR DESCRIPTION
**Pull request by soburi**
Adds notes on changes about rpi_pico

### CODE REVIEW

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -559,9 +566,12 @@ Drivers and Sensors
 
   * ``uart_emul``: added support for interrupt API.
 
+  * ``uart_rpi_pico``: Fix handling Modbus DE-RE signal
```
Comment (jhedberg) [inline doc/releases/release-notes-3.5.rst:569]: Please use the same past tense as for all other entries, i.e. "Fixed ..."

**Review by jhedberg — CHANGES_REQUESTED** [review]
See my inline comment

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/release-notes-3.5.rst [hunk 2]
```diff
@@ -294,6 +299,8 @@ Drivers and Sensors
   * Added :kconfig:option:`CONFIG_COUNTER_RTC_STM32_SUBSECONDS` to enable subsecond as
     the basic time tick on STM32 RTC based counter driver.
 
+  * Added support for RaspberryPi Pico Timer
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:302]:
```suggestion
  * Added support for Raspberry Pi Pico Timer
```

File: doc/releases/release-notes-3.5.rst [hunk 3]
```diff
@@ -559,9 +566,12 @@ Drivers and Sensors
 
   * ``uart_emul``: added support for interrupt API.
 
+  * ``uart_rpi_pico``: fixed handling Modbus DE-RE signal
+
 * SPI
 
   * Remove npcx spi driver implemented by Flash Interface Unit (FIU) module.
+  * Added support for RaspberryPi Pico PIO based SPI.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:574]:
```suggestion
  * Added support for Raspberry Pi Pico PIO based SPI.
```

**Review by kartben — APPROVED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
sorry, meant to request changes, not approve, to address small "Raspberry Pi" typo

File: doc/releases/release-notes-3.5.rst (hunk 2 above)
Comment (soburi) [reply doc/releases/release-notes-3.5.rst]: Update done.

File: doc/releases/release-notes-3.5.rst (hunk 3 above)
Comment (soburi) [reply doc/releases/release-notes-3.5.rst]: Update done.

**Review by jhedberg — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**